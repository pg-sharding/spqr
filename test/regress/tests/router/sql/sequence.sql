\c spqr-console

CREATE REFERENCE TABLE test AUTO INCREMENT id;
CREATE REFERENCE TABLE test2 AUTO INCREMENT id START 19;
CREATE REFERENCE TABLE test3 AUTO INCREMENT a START 101;

SHOW sequences;

\c regress

CREATE TABLE test(id int, age int);
CREATE TABLE test2(id int, age int);
CREATE TABLE test3(a INT, b INT, c INT);


INSERT INTO test(age) VALUES (16) /* __spqr__engine_v2: true */;
INSERT INTO test(age) VALUES (17) /* __spqr__engine_v2: true */;
INSERT INTO test(age) VALUES (18) /* __spqr__engine_v2: true */;

SELECT * FROM test /* __spqr__execute_on: sh2 */ ORDER BY id, age;

INSERT INTO test2(age) VALUES (16) /* __spqr__engine_v2: true */;
INSERT INTO test2(age) VALUES (17) /* __spqr__engine_v2: true */;
INSERT INTO test2(age) VALUES (18) /* __spqr__engine_v2: true */;

-- asserting error
INSERT INTO test2 SELECT 1 /* __spqr__engine_v2: true */;

SELECT * FROM test2 /* __spqr__execute_on: sh2 */ ORDER BY id, age;


INSERT INTO test3(b,c) VALUES (1001, 1002),(101, 102), (999, 998) /* __spqr__engine_v2: true */;

SELECT * FROM test3 /* __spqr__execute_on: sh4 */ ORDER BY 1,2,3;

DROP TABLE test;
DROP TABLE test2;
DROP TABLE test3;

\c spqr-console

DROP REFERENCE RELATION test;
DROP REFERENCE RELATION test2;
DROP REFERENCE RELATION test3;

SHOW sequences;

DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
