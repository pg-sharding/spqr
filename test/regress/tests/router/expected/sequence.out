\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE REFERENCE TABLE test AUTO INCREMENT id;
   create reference table    
-----------------------------
 table    -> test
 shard id -> sh1,sh2,sh3,sh4
(2 rows)

CREATE REFERENCE TABLE test2 AUTO INCREMENT id START 19;
   create reference table    
-----------------------------
 table    -> test2
 shard id -> sh1,sh2,sh3,sh4
(2 rows)

CREATE REFERENCE TABLE test3 AUTO INCREMENT a START 101;
   create reference table    
-----------------------------
 table    -> test3
 shard id -> sh1,sh2,sh3,sh4
(2 rows)

SHOW sequences;
   name   | value 
----------+-------
 test2_id | 19
 test3_a  | 101
 test_id  | 0
(3 rows)

\c regress
CREATE TABLE test(id int, age int);
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
CREATE TABLE test2(id int, age int);
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
CREATE TABLE test3(a INT, b INT, c INT);
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO test(age) VALUES (16) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO test(age) VALUES (17) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO test(age) VALUES (18) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
SELECT * FROM test /* __spqr__execute_on: sh2 */ ORDER BY id, age;
NOTICE: send query to shard(s) : sh2
 id | age 
----+-----
  1 |  16
  2 |  17
  3 |  18
(3 rows)

INSERT INTO test2(age) VALUES (16) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO test2(age) VALUES (17) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO test2(age) VALUES (18) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
-- asserting error
INSERT INTO test2 SELECT 1 /* __spqr__engine_v2: true */;
ERROR:  client processing error: error processing query 'INSERT INTO test2 SELECT 1 /* __spqr__engine_v2: true */;': too complex query to route, tx status IDLE
SELECT * FROM test2 /* __spqr__execute_on: sh2 */ ORDER BY id, age;
NOTICE: send query to shard(s) : sh2
 id | age 
----+-----
 20 |  16
 21 |  17
 22 |  18
(3 rows)

INSERT INTO test3(b,c) VALUES (1001, 1002),(101, 102), (999, 998) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
SELECT * FROM test3 /* __spqr__execute_on: sh4 */ ORDER BY 1,2,3;
NOTICE: send query to shard(s) : sh4
  a  |  b   |  c   
-----+------+------
 102 | 1001 | 1002
 103 |  101 |  102
 104 |  999 |  998
(3 rows)

DROP TABLE test;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
DROP TABLE test2;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
DROP TABLE test3;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP REFERENCE RELATION test;
 drop reference table 
----------------------
 table -> test
(1 row)

DROP REFERENCE RELATION test2;
 drop reference table 
----------------------
 table -> test2
(1 row)

DROP REFERENCE RELATION test3;
 drop reference table 
----------------------
 table -> test3
(1 row)

SHOW sequences;
   name   | value 
----------+-------
(0 rows)

DROP DISTRIBUTION ALL CASCADE;
       drop distribution       
-------------------------------
 distribution id -> REPLICATED
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

